### PR TITLE
Add ajax support for @AdminSecurity annotation

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | With this PR you can use `@AdminSecurity` annotation for ajax actions.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partially https://github.com/PrestaShop/PrestaShop/issues/12083
| How to test?  | This feature is currently used for toggling status on and off in list on Categories page only. :warning: Rebuild assets before testing.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13236)

### CODE REVIEW

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
How do we QA 🤔?

**Comment by sarjon** [comment]
> How do we QA thinking?

Not sure how, access control should be working as it was before, so... :thinking: Don't know how else it could be tested.

**Comment by sarjon** [comment]
@matks anything else for this PR?

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php [hunk 1]
```diff
@@ -398,12 +405,16 @@ public function deleteMenuThumbnailAction(Request $request, $categoryId, $menuTh
     /**
      * Toggle category status.
      *
-     * @param Request $request
+     * @AdminSecurity(
+     *     "is_granted(['update', 'delete'], request.get('_legacy_controller'))",
```
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php]:
```suggestion
     *     "is_granted(['update'], request.get('_legacy_controller'))",
```
This controller does not allow to delete a Category 🤔?

Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php]: Oh, I see, it's how it has been written. Mmmm ... wdyt ? I think we can remove the Delete here.

**Comment by matks** [comment]
Did the QA myself 😄

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php (hunk 1 above)
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php]: Done.

**Comment by sarjon** [comment]
@matks comment addressed. :+1:

**Review by matks — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @sarjon

**Pull request merged**